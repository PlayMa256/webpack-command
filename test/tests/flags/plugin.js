const { apply, build, crcDist, test, validate } = require('../../util');

const fixture = 'plugin/plugin';
const opts = { fixture };

let config;

test('--plugin', module, () => {
  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build`, () =>
    build(config).then((result) => {
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});
