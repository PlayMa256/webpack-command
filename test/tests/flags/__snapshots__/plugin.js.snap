// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --plugin > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    UglifyJsPlugin {
      "options": Object {
        "cache": false,
        "exclude": undefined,
        "extractComments": false,
        "include": undefined,
        "parallel": false,
        "sourceMap": false,
        "test": /\\\\\\.js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
        "uglifyOptions": Object {
          "compress": Object {
            "inline": 1,
          },
          "output": Object {
            "comments": /\\^\\\\\\*\\*!\\|@preserve\\|@license\\|@cc_on/,
          },
        },
        "warningsFilter": [Function],
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --plugin > should build #0 1`] = `
"Version: webpack 4.12.1
  Asset      Size  Chunks             Chunk Names
main.js  1.27 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --plugin > should build #1 1`] = `"406b140a"`;
